packages/playwright-core/src/server/input.ts

```diff
@@ -241,10 +279,38 @@ const aliases = new Map<string, string[]>([
   ['Enter', ['\n', '\r']],
 ]);
 
-const usKeyboardLayout = buildLayoutClosure(keyboardLayout.USKeyboardLayout);
+const defaultKeyboard = _buildLayoutClosure(defaultKeyboardLayout);
+const cache = new Map<string, KeyboardLayoutClosure>(
+    // initialized with the default keyboard layout
+    [[defaultKeyboardLayoutName, defaultKeyboard]]
+);
+
+function getByKeyboardLayoutName(layoutName?: string): KeyboardLayoutClosure {
+  if (!layoutName) return defaultKeyboard;
+
+  const normalizedLayoutName = normalizeLayoutName(layoutName);
+  const klid = keyboardLayoutNamesMapping.get(normalizedLayoutName);
+  if (!klid) throw new Error(`Keyboard layout name "${layoutName}" not found`);
+
+  const cached = cache.get(klid);
+  if (cached) return cached;
+
+  const layout: KeyboardLayout = require(`./keyboards/layouts/${klid}`).default;
+  assert(layout, `No layout found for klid ${klid}`);
```

Comment:
This should never happen, so I'd remove the `assert`. If anything, there would be an NPE down the road.